lib/events/roomevent.h

```diff
@@ -60,11 +60,15 @@ class RoomEvent : public Event {
     //! callback for that in RoomEvent.
     void addId(const QString& newId);
 
+    void setOriginalEvent(event_ptr_tt<RoomEvent> originalEvent);
+    const QJsonObject encryptedJson() const;
+
 protected:
     void dumpTo(QDebug dbg) const override;
 
 private:
     event_ptr_tt<RedactionEvent> _redactedBecause;
+    event_ptr_tt<RoomEvent> _originalEvent;
```

Comment:
`event_ptr_tt<EncryptedEvent>` for now, maybe?

Reply:
I'm not sure how to do that in the correct way, since i *have* an `event_ptr_tt<RoomEvent>`and don't know if/how i can cast that to `event_ptr_tt<EncryptedEvent>`

Reply:
Ah, I see the problem now. Yeah, static casts through unique pointers are clumsy. Let's park that for now.